Polish mobile toolbar

jasmussen opened the pull request:
This is very much an incremental and _iterative_ improvement to the mobile toolbar & experience. It is not drastic at all, and further improvement is definitely needed. But it is one step on a path. It does this a few things.

Firstly, it adds a right border to the block toolbar, when that scrolls. Before you could get this:

<img width="313" alt="screen shot 2018-06-12 at 09 31 18" src="https://user-images.githubusercontent.com/1204802/41278244-ba9ac298-6e29-11e8-8d00-83acf0e2f7af.png">

Now you get this:

<img width="331" alt="screen shot 2018-06-12 at 09 41 44" src="https://user-images.githubusercontent.com/1204802/41278249-bdfd408c-6e29-11e8-84f4-dccdb52ca14e.png">

The above is still not ideal by any means, but it's, again, an iterative improvement.

It also removes transforms from the Block's "More" menu. I _think_ that fixes #6449.

It polishes the mobile toolbar, and rearranges it so formatting is before mover and ellipsis:

<img width="429" alt="screen shot 2018-06-12 at 10 10 36" src="https://user-images.githubusercontent.com/1204802/41278330-eea2bd7a-6e29-11e8-95ac-07518a1d71fc.png">

This can perhaps be improved further, but keeping in mind the limitations of mobile safari this is still in the realm of _improvements we can do now_.

Finally, it removes the sibling inserter on mobile, because this interfered with tapping buttons, and it isn't that useful anyway given that you can see a "Add block" button right in the bottom toolbar.

Commit:
Add right border when block toolbar scrolls.

Commit:
Remove transformations from Block More menu.

Commit:
Polish mobile toolbar.

Commit:
Don't show sibling inserter on mobile.

tofumatt commented on edit-post/assets/stylesheets/_variables.scss:
```diff
@@ -39,7 +39,7 @@ $block-controls-height: 36px;
 $icon-button-size-small: 24px;
 $inserter-tabs-height: 36px;
-$block-toolbar-height: 37px;
+$block-toolbar-height: $block-controls-height + 1px;
```
I get why this is but a comment to explain it would be great 😄

tofumatt commented on editor/components/block-list/style.scss:
```diff
@@ -581,17 +581,27 @@
-		margin-top: $item-spacing;
+		margin-top: $item-spacing + $block-toolbar-height; // Make room for the formatting bar above. 
+		margin-right: -$block-padding;
+		margin-bottom: -$block-padding - 1px; // Include border.
```
If the border is the `1px` value we should be storing `$border = 1px;` somewhere or when we update one we might forget to update the other. Also then it's just easier/automatic.

tofumatt commented on editor/components/block-list/style.scss:
```diff
@@ -581,17 +581,27 @@
 		display: flex;
 		flex-direction: row;
-		margin-top: $item-spacing;
+		margin-top: $item-spacing + $block-toolbar-height; // Make room for the formatting bar above. 
```
This comment helps, but `// Make room for the height of the block toolbar above.` is even more explicit/helpful.

tofumatt commented on editor/components/block-list/style.scss:
```diff
@@ -581,17 +581,27 @@
+		margin-right: -$block-padding;
+		margin-bottom: -$block-padding - 1px; // Include border.
+		margin-left: -$block-padding;
+		border-top: 1px solid $light-gray-500;
```
Yeah, looks like this `1px` could be a variable, that'd help with the `// include border` comment above.

tofumatt commented on editor/components/block-list/style.scss:
```diff
@@ -581,17 +581,27 @@
+			box-shadow: none;
+		}
+
+		// Movers, inserter, trash & ellipsis.
```
Don't we use Oxford commas in WordPress style? This should be `// Movers, inserter, trash, and ellipsis.`

tofumatt commented on editor/components/block-list/style.scss:
```diff
@@ -753,18 +768,17 @@
 	// Position toolbar below the block on mobile
 	position: absolute;
-	bottom: -$block-toolbar-height + 1px;
+	//bottom: -$block-toolbar-height + 1px;
```
🔥

tofumatt commented on editor/components/block-toolbar/style.scss:
```diff
@@ -18,7 +18,6 @@
 		border-right: 1px solid $light-gray-500;
 	}
 
-	// this should probably have its own class
```
Is this actually no longer the case? I'd still agree with the comment, I don't like seeing selectors like `> div`; they seem sloppy.

jasmussen replied:
My thinking in removing this comment was perhaps fast and loose. I think it's still the case that it should be a class, but I this not happening in the near future, and instead of leaving a "note to self" might as well remove it. I remember trying to add the class, but not finding a clean way to do it.

tofumatt replied:
All of that sounds useful to know, and rather like the comment should be expanded upon rather than deleted 😄

tofumatt reviewed (changes requested):
I think there are a few cleanups needed here.

jasmussen commented:
Solid review points. I'll address them all!

Commit:
Address feedback and fix typo.

The typo was a actually committed in https://github.com/WordPress/gutenberg/pull/6408/files but this fixes it.

jasmussen commented:
I think I caught all your points. Additionally I found and fixed a typo from an old PR. Care to take another look? Thank you!

tofumatt commented:
Sure thing! In the future, you can request a re-review by selecting me for review again:

![2018-06-13 10 10 21](https://user-images.githubusercontent.com/90871/41341609-07d48c92-6ef2-11e8-8ade-3fb68a13ec17.gif)

Doing it is more of a pain because the UI isn't amazing, but it means it shows up in my pull request queue as a review request assigned to me, which is easier to find and means I won't miss it, unlike a comment 😄

jasmussen commented:
> Doing it is more of a pain because the UI isn't amazing, but it means it shows up in my pull request queue as a review request assigned to me, which is easier to find and means I won't miss it, unlike a comment 😄

mindblown.webm

👍 👍

tofumatt commented on editor/components/block-list/style.scss:
```diff
@@ -391,7 +390,7 @@
 		> .editor-block-mover,
 		> .editor-block-settings-menu {
-			top: -$block-side-ui-width - 1px; // move upwards the height of the button +1px for border
+			top: -$block-side-ui-width - $border-width; // Move upwards the height of the button & border.
```
This is phrased a bit oddly. How about:

```scss
top: -$block-side-ui-width - $border-width; // This moves the menu up by the height of the button + border.
```

tofumatt commented on editor/components/block-list/style.scss:
```diff
@@ -751,51 +765,48 @@
-	// put toolbar snugly to side edges on mobile
-	margin-left: -$block-padding - 1px;	// stack borders
-	margin-right: -$block-padding - 1px;
+	// Put toolbar snugly to side edges on mobile.
```
"Snugly" is a cute turn of phrase but is a bit of a colloquialism. Maybe there's a better way to phrase this for less adapt English folk?

tofumatt commented on editor/components/block-list/style.scss:
```diff
@@ -751,51 +765,48 @@
 		}
 
-		// except for wide elements, this causes a horizontal scrollbar
+		// ... Except for wide elements, this causes a horizontal scrollbar.
```
*Technically* after an ellipsis you probably *don't* want a capital letter. That said there's no preceding ellipsis here so I'd just remove the ellipsis.

tofumatt reviewed (approved):
Looks good to me; minor tweaks needed but after they're done I say it's good once the tests pass.

Commit:
Address feedback on comments.
